app/jobs/irs_attempts_events_batch_job.rb

```diff
@@ -27,14 +31,31 @@ def perform(timestamp = Time.zone.now - 1.hour)
       filename: result.filename,
       iv: encoded_iv,
       encrypted_key: encoded_encrypted_key,
-      requested_time: IrsAttemptsApi::EnvelopeEncryptor.formatted_timestamp(timestamp),
+      requested_time: timestamp_key(key: timestamp),
     )
 
     log_irs_attempts_events_job_info(result, events, start_time)
     redis_client.remove_events(timestamp: timestamp) if upload_to_s3_response&.etag
     irs_attempts_api_log_file
   end
 
+  def timestamp_key(key:)
+    IrsAttemptsApi::EnvelopeEncryptor.formatted_timestamp(key)
+  end
+
+  def missing_log_file?(previous_hour)
+    previous_hour_log_file_hash = {
+      requested_time: timestamp_key(key: previous_hour),
+    }
+
+    !IrsAttemptApiLogFile.find_by(previous_hour_log_file_hash) &&
+      reasonable_timespan(previous_hour)
+  end
+
+  def reasonable_timespan(check_time)
```

Comment:
I didn't think of this when we're pairing, and wouldn't be upset if you just merged this without incorporating it, but I might tack a question mark on the end and make this `reasonable_timespan?(check_time)` so it's more clear that it's checking rather than returning a time.